employee_seniority_level/models/account_analytic_line.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, models, _
+from odoo.exceptions import UserError
+
+
+class AccountAnalyticLine(models.Model):
+
+    _inherit = 'account.analytic.line'
+
+    @api.model
+    def create(self, vals):
+        employee = None
+        if 'employee_id' in vals:
+            employee = self.env['hr.employee'].browse(vals['employee_id'])
+        elif self.env.user.employee_ids:
+            employee = self.env.user.employee_ids[0]
+
+        if employee and 'project_id' in vals:
+            if not employee.seniority_level_id:
```

Comment:
upon reflection, we need to change the spec a little bit: if there are no products with a seniority level in the sale order, then timesheeting is allowed and we do nothing special.

Reply:
implemented in https://github.com/TDu/project/pull/1 with other fixes